gtest link error with gcc >= 6.1.0

Compiling gtest with g++ >= 6.1.0 yields the following link error. Using g++ 5.3.0 (or lesser) works. 
```
/bin/sh ../../libtool  --tag=CXX   --mode=link g++  -g -O3 -fno-tree-vectorize -O3 -g -Wall -Werror   -no-install  -o gtest common/gtest-gtest-all.o common/gtest-main.o common/gtest-test_helpers.o common/gtest-t
est_perf.o common/gtest-test.o ucm/gtest-malloc_hook.o uct/gtest-test_amo.o uct/gtest-test_amo_add.o uct/gtest-test_amo_cswap.o uct/gtest-test_amo_fadd.o uct/gtest-test_amo_swap.o uct/gtest-test_fence.o uct/gtes
t-test_flush.o uct/gtest-test_many2one_am.o uct/gtest-test_mm.o uct/gtest-test_mem.o uct/gtest-test_p2p_am.o uct/gtest-test_p2p_err.o uct/gtest-test_p2p_rma.o uct/gtest-test_pd.o uct/gtest-test_pending.o uct/gte
st-test_uct_ep.o uct/gtest-test_uct_perf.o uct/gtest-uct_p2p_test.o uct/gtest-uct_test.o uct/gtest-test_stats.o uct/gtest-test_wakeup.o uct/gtest-test_error_handling.o ucp/gtest-test_ucp_atomic.o ucp/gtest-test_
ucp_memheap.o ucp/gtest-test_ucp_mmap.o ucp/gtest-test_ucp_perf.o ucp/gtest-test_ucp_rma.o ucp/gtest-test_ucp_tag_cancel.o ucp/gtest-test_ucp_tag_match.o ucp/gtest-test_ucp_tag_probe.o ucp/gtest-test_ucp_tag_xfe
r.o ucp/gtest-test_ucp_tag.o ucp/gtest-test_ucp_context.o ucp/gtest-test_ucp_wireup.o ucp/gtest-test_ucp_wakeup.o ucp/gtest-test_ucp_fence.o ucp/gtest-ucp_test.o ucs/gtest-test_algorithm.o ucs/gtest-test_arbiter
.o ucs/gtest-test_async.o ucs/gtest-test_callbackq.o ucs/gtest-test_class.o ucs/gtest-test_component.o ucs/gtest-test_config.o ucs/gtest-test_datatype.o ucs/gtest-test_debug.o ucs/gtest-test_memtrack.o ucs/gtest
-test_instr.o ucs/gtest-test_math.o ucs/gtest-test_mpmc.o ucs/gtest-test_mpool.o ucs/gtest-test_pgtable.o ucs/gtest-test_rcache.o ucs/gtest-test_stats.o ucs/gtest-test_sys.o ucs/gtest-test_time.o ucs/gtest-test_
twheel.o ucs/gtest-test_frag_list.o ucs/gtest-test_hash_perf.o uct/gtest-test_ib.o uct/gtest-ud_base.o uct/gtest-test_ud.o uct/gtest-test_ud_slow_timer.o uct/gtest-test_ud_pending.o uct/gtest-test_ud_ds.o uct/gt
est-test_rc.o  ../../src/ucs/libucs.la ../../src/uct/libuct.la ../../src/ucm/libucm.la ../../src/ucp/libucp.la ../../src/tools/perf/libucxperf.la  -lz -lpthread -lrt -lrt  -lbfd -liberty -ldl
libtool: link: g++ -g -O3 -fno-tree-vectorize -O3 -g -Wall -Werror -o gtest common/gtest-gtest-all.o common/gtest-main.o common/gtest-test_helpers.o common/gtest-test_perf.o common/gtest-test.o ucm/gtest-malloc_
hook.o uct/gtest-test_amo.o uct/gtest-test_amo_add.o uct/gtest-test_amo_cswap.o uct/gtest-test_amo_fadd.o uct/gtest-test_amo_swap.o uct/gtest-test_fence.o uct/gtest-test_flush.o uct/gtest-test_many2one_am.o uct/
gtest-test_mm.o uct/gtest-test_mem.o uct/gtest-test_p2p_am.o uct/gtest-test_p2p_err.o uct/gtest-test_p2p_rma.o uct/gtest-test_pd.o uct/gtest-test_pending.o uct/gtest-test_uct_ep.o uct/gtest-test_uct_perf.o uct/g
test-uct_p2p_test.o uct/gtest-uct_test.o uct/gtest-test_stats.o uct/gtest-test_wakeup.o uct/gtest-test_error_handling.o ucp/gtest-test_ucp_atomic.o ucp/gtest-test_ucp_memheap.o ucp/gtest-test_ucp_mmap.o ucp/gtes
t-test_ucp_perf.o ucp/gtest-test_ucp_rma.o ucp/gtest-test_ucp_tag_cancel.o ucp/gtest-test_ucp_tag_match.o ucp/gtest-test_ucp_tag_probe.o ucp/gtest-test_ucp_tag_xfer.o ucp/gtest-test_ucp_tag.o ucp/gtest-test_ucp_
context.o ucp/gtest-test_ucp_wireup.o ucp/gtest-test_ucp_wakeup.o ucp/gtest-test_ucp_fence.o ucp/gtest-ucp_test.o ucs/gtest-test_algorithm.o ucs/gtest-test_arbiter.o ucs/gtest-test_async.o ucs/gtest-test_callbac
kq.o ucs/gtest-test_class.o ucs/gtest-test_component.o ucs/gtest-test_config.o ucs/gtest-test_datatype.o ucs/gtest-test_debug.o ucs/gtest-test_memtrack.o ucs/gtest-test_instr.o ucs/gtest-test_math.o ucs/gtest-te
st_mpmc.o ucs/gtest-test_mpool.o ucs/gtest-test_pgtable.o ucs/gtest-test_rcache.o ucs/gtest-test_stats.o ucs/gtest-test_sys.o ucs/gtest-test_time.o ucs/gtest-test_twheel.o ucs/gtest-test_frag_list.o ucs/gtest-te
st_hash_perf.o uct/gtest-test_ib.o uct/gtest-ud_base.o uct/gtest-test_ud.o uct/gtest-test_ud_slow_timer.o uct/gtest-test_ud_pending.o uct/gtest-test_ud_ds.o uct/gtest-test_rc.o  ../../src/ucs/.libs/libucs.so ../
../src/uct/.libs/libuct.so ../../src/ucm/.libs/libucm.so ../../src/ucp/.libs/libucp.so ../../src/tools/perf/.libs/libucxperf.a /home/bouteill/ucx/debug.build/src/ucp/.libs/libucp.so /home/bouteill/ucx/debug.buil
d/src/uct/.libs/libuct.so -libverbs -libcm /home/bouteill/ucx/debug.build/src/ucs/.libs/libucs.so /home/bouteill/ucx/debug.build/src/ucm/.libs/libucm.so /opt/gcc/6.2.0/lib/../lib64/libstdc++.so -lm -lz -lpthread
 -lrt -lbfd -liberty -ldl -Wl,-rpath -Wl,/home/bouteill/ucx/debug.build/src/ucs/.libs -Wl,-rpath -Wl,/home/bouteill/ucx/debug.build/src/uct/.libs -Wl,-rpath -Wl,/home/bouteill/ucx/debug.build/src/ucm/.libs -Wl,-
rpath -Wl,/home/bouteill/ucx/debug.build/src/ucp/.libs -Wl,-rpath -Wl,/opt/gcc/6.2.0/lib/../lib64 -Wl,-rpath -Wl,/home/bouteill/ucx/debug.build/lib -Wl,-rpath -Wl,/opt/gcc/6.2.0/lib/../lib64
ucp/gtest-test_ucp_wireup.o: In function `std::vector<unsigned long, std::allocator<unsigned long> >::insert(__gnu_cxx::__normal_iterator<unsigned long const*, std::vector<unsigned long, std::allocator<unsigned 
long> > >, unsigned long, unsigned long const&)':
/opt/gcc/6.1.0/include/c++/6.1.0/bits/stl_vector.h:1055: undefined reference to `test_ucp_wireup::SEND_DATA'
collect2: error: ld returned 1 exit status
```

--- a/test/gtest/ucp/test_ucp_wireup.cc
+++ b/test/gtest/ucp/test_ucp_wireup.cc
@@ -92,7 +92,7 @@ test_ucp_wireup::enum_test_params(const ucp_params_t& ctx_params,
 void test_ucp_wireup::init() {
     ucp_test::init();
 
-    m_send_data.resize(BUFFER_LENGTH, SEND_DATA);
+    m_send_data.resize(BUFFER_LENGTH, elem_type(SEND_DATA));
     m_recv_data.resize(BUFFER_LENGTH, 0);
 
     if (GetParam().variant == UCP_FEATURE_RMA) {

